pkg/operator/apiserver/controller/apiservice/apiservice_controller.go

```diff
@@ -23,8 +24,20 @@ import (
 	"github.com/openshift/library-go/pkg/operator/v1helpers"
 )
 
-type GetAPIServicesToMangeFunc func() ([]*apiregistrationv1.APIService, error)
-type apiServicesPreconditionFuncType func([]*apiregistrationv1.APIService) (bool, error)
+// ManagedAPIService keeps expected state of a single APIService object
+type ManagedAPIService struct {
+	// Enabled signals whether an APIService is enabled.
+	// Any disabled APIService is removed from a cluster.
+	Enabled bool
```

Comment:
will this field be used by all clients or only for some?

Reply:
Once such struct is created, it's intended to be used only by the controller.

Reply:
An alternative would be to provide a filter function that if not provide will include every api service. On the other hand such a filter will get invoked all over again instead of constructing a list of enabled/disabled service only once at the beginning.

Reply:
OK, since there are only two operators using this controller I would:

1. Change the signature of the method to `GetAPIServicesToMangeFunc func() (enabled []*apiregistrationv1.APIService, disabled []*apiregistrationv1.APIService, error)`
2. Rename `precondition` to `preconditionForEnabledAPIServices`
3. Split the `sync` method into two flows, one for reconciling `enabled` services and the second for reconciling `disabled` services.

Does it make sense ?

Reply:
Done. It looks better now.

Reply:
Thanks, I have added a few more comments/questions, PTAL.